bridges/relays/lib-substrate-relay/src/parachains/mod.rs

```diff
@@ -74,37 +68,3 @@ pub trait SubmitParachainHeadsCallBuilder<P: SubstrateParachainsPipeline>:
 		is_free_execution_expected: bool,
 	) -> CallOf<P::TargetChain>;
 }
-
-/// Building `submit_parachain_heads` call when you have direct access to the target
-/// chain runtime.
-pub struct DirectSubmitParachainHeadsCallBuilder<P, R, I> {
-	_phantom: PhantomData<(P, R, I)>,
-}
-
-impl<P, R, I> SubmitParachainHeadsCallBuilder<P> for DirectSubmitParachainHeadsCallBuilder<P, R, I>
-where
-	P: SubstrateParachainsPipeline,
-	P::SourceRelayChain: Chain<Hash = RelayBlockHash, BlockNumber = RelayBlockNumber>,
-	R: BridgeParachainsConfig<I> + Send + Sync,
-	I: 'static + Send + Sync,
-	R::BridgedChain: bp_runtime::Chain<
-		BlockNumber = RelayBlockNumber,
-		Hash = RelayBlockHash,
-		Hasher = RelayBlockHasher,
-	>,
-	CallOf<P::TargetChain>: From<BridgeParachainsCall<R, I>>,
-{
-	fn build_submit_parachain_heads_call(
-		at_relay_block: HeaderIdOf<P::SourceRelayChain>,
-		parachains: Vec<(ParaId, ParaHash)>,
-		parachain_heads_proof: ParaHeadsProof,
-		_is_free_execution_expected: bool,
-	) -> CallOf<P::TargetChain> {
-		BridgeParachainsCall::<R, I>::submit_parachain_heads {
-			at_relay_block: (at_relay_block.0, at_relay_block.1),
-			parachains,
-			parachain_heads_proof,
-		}
-		.into()
-	}
-}
```

Comment:
Can we keep this ? I think it will be needed for https://github.com/paritytech/parity-bridges-common/issues/2952

Reply:
Makes sense - I've reverted removal. I've also recently removed similar struct for GRANDPA submissions - you may need it as well.